src/servers/vsgi-http.vala

```diff
@@ -240,36 +236,34 @@ namespace VSGI.HTTP {
 		public bool init (Cancellable? cancellable = null) throws GLib.Error {
 			if (https) {
 				server = new Soup.Server (
-					Soup.SERVER_RAW_PATHS,       raw_paths,
-					Soup.SERVER_TLS_CERTIFICATE, tls_certificate);
+					"raw-paths", raw_paths,
+					"tls-certificate", tls_certificate);
 			} else {
-				server = new Soup.Server (
-					Soup.SERVER_RAW_PATHS,       raw_paths,
-					Soup.SERVER_TLS_CERTIFICATE, tls_certificate);
+				server = new Soup.Server ("raw-paths", raw_paths);
 			}
 
 			// register a catch-all handler
-			server.add_handler (null, (server, msg, path, query, client) => {
-				msg.set_status (Soup.Status.OK);
+			server.add_handler (null, (server, msg, path, query) => {
+				msg.set_status (Soup.Status.OK, null);
 
 				// prevent I/O as we handle everything asynchronously
 				server.pause_message (msg);
 
-				var req = new Request (msg, client, query);
+				var req = new Request (msg, query);
 				var res = new Response (req, server, msg);
 
 				var auth = req.headers.get_one ("Authorization");
-				if (auth != null) {
-					if (Soup.str_case_equal (auth.slice (0, 6), "Basic ")) {
-						var auth_data = (string) Base64.decode (auth.substring (6));
-						if (auth_data.index_of_char (':') != -1) {
-							req.uri.set_user (auth_data.slice (0, auth.index_of_char (':')));
-						}
-					} else if (Soup.str_case_equal (auth.slice (0, 7), "Digest ")) {
-						var auth_data = Soup.header_parse_param_list (auth.substring (7));
-						req.uri.set_user (auth_data["username"]);
-					}
-				}
+				// if (auth != null) {
+				// 	if (VSGI.str_case_equal (auth.slice (0, 6), "Basic ")) {
+				// 		var auth_data = (string) Base64.decode (auth.substring (6));
+				// 		if (auth_data.index_of_char (':') != -1) {
+				// 			req.uri.set_user (auth_data.slice (0, auth.index_of_char (':')));
+				// 		}
+				// 	} else if (VSGI.str_case_equal (auth.slice (0, 7), "Digest ")) {
+				// 		var auth_data = Soup.header_parse_param_list (auth.substring (7));
+				// 		req.uri.set_user (auth_data["username"]);
+				// 	}
+				// }
```

Comment:
Two things here: `uri` is immutable, so this is not a suitable approach and `str_case_equal` is not visible despite passing `--vapi` with the internal libvsgi Vala symbols.